.github/workflows/docstrings.yml

```diff
@@ -15,15 +15,16 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
     - name: Setup Python
       uses: actions/setup-python@v5
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install pytest
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        python -m pip install meson-python meson ninja pybind11 versioningit pytest
     - name: Install
```

Comment:
Luke, could you briefly describe what `--no-build-isolation -e` is doing here?

Reply:
I think I figured it out,
- isolates the build environment from any system packages
- ensures that build requirements are installed

And --no-build-isolation disables both.
(from Todd Gamblin)